File: 프로그래머스/1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.java
class Solution {
    public String solution(String phone_number) {
        String answer = "";
        
        for(int i = 0; i < phone_number.length() - 4; i++) {
            answer += "*";
        }
        
        answer += phone_number.substring(phone_number.length() - 4);
        
        return answer;
    }
}

// 다른 풀이
 char[] ch = phone_number.toCharArray();
 for(int i = 0; i < ch.length - 4; i ++){
     ch[i] = '*';
 }
 return String.valueOf(ch);

// 다른 풀이
return phone_number.replaceAll(".(?=.{4})", "*");

File: 프로그래머스/1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.py
def solution(phone_number):
    answer = ''
    
    idx = len(phone_number) - 4
    answer = '*' * idx
    answer += phone_number[idx:]
    
    return answer

# 다른 풀이
return "*"*(len(s)-4) + s[-4:] # -4로 뒤에서 가져오기

